routes.go

```diff
@@ -64,72 +69,88 @@ func (ctrl *Controller) EditJob(ctx *gin.Context) {
 
 func (ctrl *Controller) CreateJob(ctx *gin.Context) {
 	var newJobInput NewJob
-	ctx.Bind(&newJobInput)
+	if err := ctx.Bind(&newJobInput); err != nil {
+		log.Println(fmt.Errorf("failed to ctx.Bind: %w", err))
+		ctx.AbortWithStatus(http.StatusInternalServerError)
+		return
+	}
 
 	session := sessions.Default(ctx)
+	defer func() {
+		if err := session.Save(); err != nil {
+			log.Println(fmt.Errorf("CreateJob failed to session.Save: %w", err))
+		}
+	}()
 
 	if errs := newJobInput.validate(false); len(errs) != 0 {
 		for k, v := range errs {
 			session.AddFlash(v, fmt.Sprintf("%s_err", k))
 		}
-		session.Save()
 
 		ctx.Redirect(302, "/new")
 		return
 	}
 
 	job, err := newJobInput.saveToDB(ctrl.DB)
 	if err != nil {
-		log.Print(fmt.Errorf("failed to save job to db: %w", err))
-
+		log.Println(fmt.Errorf("failed to save job to db: %w", err))
 		session.AddFlash("Error creating job")
-		session.Save()
-
 		ctx.Redirect(302, "/new")
 		return
 	}
 
-	// TODO: make this a nicer html template?
-	message := fmt.Sprintf(
-		"Your job has been created!\n\n<a href=\"%s\">Use this link to edit the job posting</a>",
-		signedJobRoute(job, ctrl.Config),
-	)
-	err = sendEmail(newJobInput.Email, "Job Created!", message, ctrl.Config.Email)
-	if err != nil {
-		panic(err) // TODO: handle
+	if ctrl.Config.Email.SMTPHost != "" {
```

Comment:
It helped locally when I didn't have email setup. I didn't look closely enough, but it looks like we only attempt this email send at creation, and if it doesn't send, then that post is locked forever. Is there a way to request a new email? I played with returning the link to the client as a part of the Flash if the email send fails, but thought it was a bit _ugly_.

Reply:
> then that post is locked forever

Correct, however with our small scale I've accepted that as a risk 😅. I do like the idea of showing the poster the edit link once though, doing that in addition to emailing it seems like a valuable addition. The _ugly_ part can be worked on I'm sure, haha.

> Is there a way to request a new email?

Not currently. That seems like it could be valuable but also if it failed the first time (due to a mistyped email or something) it has a high chance of failing the second time as well. We would need to expose some way to provide a _new_ email, which requires verifying they were the original poster, and email is the only identity piece we have right now so 🤷 

I think your idea of showing the poster the edit link in the UI is a nice consolation for a potentially failed email send. Seems worth capturing as a separate issue!